### PR TITLE
[REVIEW][FEATURE][GAWE-50]-business logic set new pw

### PR DESCRIPTION
**Pull request by azkifairuz**
#### Description

> Description of features, bugs, improvements or any changes in this MR.

#### Ticket Reference(s)

[GAWE-50](https://bluehabit.atlassian.net/browse/GAWE-50)

##### UI Changes

> Screenshot or video of UI-related changes

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

> masih blm ngerti dan meraba raba

[GAWE-50]: https://bluehabit.atlassian.net/browse/GAWE-50?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import io.ktor.client.request.setBody
+import javax.inject.Inject
+
+class SetNewPasswordRepository @Inject constructor(
+    private val httpClient: HttpClient
+) {
+    suspend fun execute(password: String):Response<SetNewPasswordResponse>{
+        val result = safeApiCall<SetNewPasswordResponse> {
+            httpClient.post(AuthApi.SetNewPassword()) {
+                setBody(
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt:26]: set header nya disini

**Review by triandamai — CHANGES_REQUESTED** [review]
cek

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt (hunk 1 above)
Comment (triandamai) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt:26]:
```kotlin

client.post("") {
    headers {
        append(HttpHeaders.Accept, "text/html")
        append(HttpHeaders.Authorization, "abc123")
        append(HttpHeaders.UserAgent, "ktor client")
    }
    setBody(
      //ini body
    )
}
```

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/AuthConstant.kt [hunk 2]
```diff
@@ -9,4 +9,5 @@ package com.bluehabit.eureka.data.authentication
 
 object AuthConstant {
     const val SESSION_OTP_ID = "sessionOtpId"
+    const val SESSION_TOKEN_RESET_PW = "sessionToken"
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/AuthConstant.kt:12]: jangan disingkat langsun PASSWORD aja

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt [hunk 3]
```diff
@@ -7,22 +7,35 @@
...
-    private val httpClient: HttpClient
+    private val httpClient: HttpClient,
+    private val pref: SharedPref
 ) {
     suspend fun execute(password: String):Response<SetNewPasswordResponse>{
         val result = safeApiCall<SetNewPasswordResponse> {
             httpClient.post(AuthApi.SetNewPassword()) {
+                headers {
+                    append(HttpHeaders.Accept, "text/html")
+                    append(HttpHeaders.Authorization, "abc123")
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt:35]: ga perlu Authorization,Accept,UserAgent

**Review by triandamai — CHANGES_REQUESTED** [review]

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt [hunk 4]
```diff
@@ -31,10 +28,7 @@ class SetNewPasswordRepository @Inject constructor(
         val result = safeApiCall<SetNewPasswordResponse> {
             httpClient.post(AuthApi.SetNewPassword()) {
                 headers {
-                    append(HttpHeaders.Accept, "text/html")
-                    append(HttpHeaders.Authorization, "abc123")
-                    append(HttpHeaders.UserAgent, "ktor client")
-                    append("4adf-3ed", pref.getPersistData(AuthConstant.SESSION_TOKEN_RESET_PW).orEmpty())
+                    append("4adf-3ed", pref.getPersistData(AuthConstant.SESSION_TOKEN_RESET_PASSWORD).orEmpty())
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt:31]: ini taro di constanta aja @azkifairuz

**Review by triandamai — CHANGES_REQUESTED** [review]
cek dulu

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt (hunk 4 above)
Comment (azkifairuz) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt:31]: yg token "4asnkancjk" nya
?

Comment (triandamai) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/SetNewPasswordRepository.kt:31]: yang '4adf-3ed' @azkifairuz 
taro di `AuthConstant` aja misal `const val keyHeaderSessionId = "4adf-3ed"`

**Review by triandamai — APPROVED** [review]

**Pull request merged**